float32 steering
float32 finalAngle